test/python/compiler/test_assembler.py

```diff
@@ -571,6 +576,19 @@ def test_assemble_schedule_enum(self):
         self.assertEqual(test_dict['config']['meas_return'], 'avg')
         self.assertEqual(test_dict['config']['meas_level'], 2)
 
+    def test_instruction_validation(self):
+        """Test instruction validation according to meas map"""
+        acquire = pulse.Acquire(2)
+
+        instructions = [AcquireInstruction(acquire, AcquireChannel(i), MemorySlot(i))
+                        for i in range(2)]
+        self.assertTrue(_validate_meas_map(instructions, meas_map=[[0, 1]]))
+
+        instructions = [AcquireInstruction(acquire, AcquireChannel(0), MemorySlot(i))
+                        for i in range(2)]
+        # TODO: self.assertRaises(QiskitError, _validate_meas_map, instructions, [[0]])
```

Comment:
The syntax for this isn't obvious
```suggestion
        with self.assertRaises(QiskitError):
            _validate_meas_map(instructions, meas_map=[[0]])
```

Reply:
I decided to remove it at all, due to it is out of scope of this issue + it is still not clear wether this case should raise error or not (because function just checks that all qubits that in meas map are acquired and not checking wether they overlap or not).

Maybe we want to revisit this function and take into consideration acquire durations and times and validate based on that.

Reply:
Fair 👍